libvirt/tests/src/virtual_network/iface_network.py

```
@@ -215,6 +215,26 @@ def run_dnsmasq_host_test(iface_mac, guest_ip, guest_name):
         if not configs.count(config):
             test.fail("Can't find host configuration in file %s" % conf_file)
 
+    def check_service_status(service, br_ip):
+        """
+        Check the service's status by the listening ports
+
+        :param service: dnsmasq provided services, dns or dhcp
+        :param br_ip: the ip of the linux bridge which dnsmasq attached
+        :return: True or False. If the service is active, return True,
+                else return False;
+        """
+        cmd = "netstat -nlp | grep dnsmasq"
+        outputs = process.run(cmd, shell=True, ignore_status=True).stdout_text
+        logging.debug("The port opened by dnsmasq:\n %s", outputs)
+        dns_port = '%s:53' % br_ip
+        dhcp_port = '0.0.0.0:67'
+        if (service == 'dns' and dns_port in outputs) or \
+                (service == 'dhcp' and dhcp_port in outputs):
+            return True
```

since the method name is check_service_status(), so you may use test.fail directly among this method, instead of return True Or False

---

The service status can be active or inactive, if using test.fail() directly among this method, one more param "expected-status" need to be added. So I don't think there are very clear benefits. What do you think?

---

But I have updated the func name to be "service_is_active", it is more understandable.